frappe_graphql/utils/generate_sdl/doctype.py

```diff
@@ -36,11 +36,14 @@ def get_basic_doctype_sdl(meta: Meta):
             fieldtype = "Int"
         elif field in ("owner", "modified_by"):
             fieldtype = "User!"
+        elif field == "parent":
+            fieldtype = "BaseDocType"
         else:
             fieldtype = "String"
         sdl += f"\n  {field}: {fieldtype}"
     sdl += "\n  owner__name: String!"
     sdl += "\n  modified_by__name: String!"
+    sdl += "\n  parent__name: String"
```

Comment:
@faztp12 best to check for child doctypes only right ?

Reply:
@e-lobo but, i believe normal DocTypes could be made a child doctype too ?

Reply:
@faztp12 i dont think so.. i believe only if you declare a doctype as a child doctype can it be one

Reply:
@assemmarwan @e-lobo 
I added parent fields to `BaseDocType` because frappe has those fields defined in db for each doctype.
Since frappe do not allow adding non-child doctype as child table, we should remove `parent` , `parenttype` & `parentfield` from `BaseDocType` interface, and introduce a new interface `BaseChildDocType` that has these parent fields. 

What do you guys think ?

Reply:
BaseDocType is enough I reckon 👍🏻